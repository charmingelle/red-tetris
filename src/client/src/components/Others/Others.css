.others {
  list-style-type: none;
  margin: 0;
  padding: 0;
  display: flex;
  flex-direction: column;
}

.other {
  margin: 10px;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.other-pile {
  position: relative;
  width: 100px;
  height: 200px;
  outline: 5px solid #999;
  background-color: black;
}

.score {
  margin-top: 10px;
  font-size: 24px;
  color: #999;
}
