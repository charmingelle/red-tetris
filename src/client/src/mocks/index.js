export const RED = {
  main: '#c00',
  lighter: '#f00',
  darker: '#980000',
};

export const LINE = [
  [0, 0, 1, 0],
  [0, 0, 1, 0],
  [0, 0, 1, 0],
  [0, 0, 1, 0],
];

export const GI = [
  [0, 0, 0],
  [1, 1, 1],
  [0, 0, 1],
];

export const EMPTY_PILE = [
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
];

export const NOT_EMPTY_PILE = [
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [RED, RED, RED, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, RED, 0, 0, 0, 0, 0, 0, 0],
];

export const GAME_OVER_PILE = [
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, RED, RED, RED, RED, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [RED, RED, RED, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, RED, 0, 0, 0, 0, 0, 0, 0],
];

export const PILE_WITH_ROWS_TO_REMOVE = [
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [RED, RED, RED, RED, RED, RED, RED, RED, RED, RED],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
  [RED, RED, RED, RED, RED, RED, RED, RED, RED, RED],
  [RED, RED, RED, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, RED, 0, 0, 0, 0, 0, 0, 0],
];

export const TETRO = {
  figure: [GI],
  row: 5,
  col: 4,
  color: RED,
};

export const ROOM = {
  id: 'Room',
  leader: 'id-0',
  players: {
    'id-0': {
      id: 'id-0',
      name: 'Anna',
      pile: EMPTY_PILE,
      tetro: null,
      score: 0,
      isGameOver: false,
    },
  },
};

export const EMPTY_STORE = {
  socket: null,
  myId: null,
  myName: null,
  myRoom: null,
  people: {},
  rooms: {},
  tetro: null,
};

export const PERSON_ANNA = { name: 'Anna', score: 0 };

export const PERSON_OLYA = { name: 'Olya', score: 0 };
