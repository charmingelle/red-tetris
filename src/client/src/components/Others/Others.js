import React from 'react';
import { connect } from 'react-redux';
import './Others.css';
import { Square } from '../Square';
import { OTHER_SHIFT } from '../../constants';

const renderOtherPile = pile => (
  <div className="other-pile">
    {pile.map((row, rowIndex) =>
      row.map((el, colIndex) =>
        el !== 0 ? (
          <Square
            left={colIndex}
            top={rowIndex}
            color={el}
            key={`${rowIndex}+${colIndex}`}
            shift={OTHER_SHIFT}
          />
        ) : null,
      ),
    )}
  </div>
);

const OthersInner = ({ others }) => (
  <ul className="others">
    {Object.keys(others).map(playerId => (
      <li key={playerId} className="other">
        {renderOtherPile(others[playerId])}
        <div className="score">{`${playerId}: 0`}</div>
      </li>
    ))}
  </ul>
);

const mapStateToProps = ({ others }) => ({ others });

export const Others = connect(mapStateToProps)(OthersInner);
