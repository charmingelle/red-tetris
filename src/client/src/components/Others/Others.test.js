import React from 'react';
import { Others } from './Others';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { RED, LINE, GI, EMPTY_PILE, NOT_EMPTY_PILE } from '../../mocks';

const mockStore = configureStore([]);

test('Others in room with 3 players', () => {
  const store = mockStore({
    socket: null,
    myId: 'id-0',
    myName: 'Anna',
    myRoom: {
      id: 'Room',
      leader: 'id-0',
      players: {
        'id-0': {
          id: 'id-0',
          name: 'Anna',
          pile: NOT_EMPTY_PILE,
          tetro: {
            figure: LINE,
            row: 1,
            col: 4,
            color: RED,
          },
          score: 0,
          isGameOver: false,
        },
        'id-1': {
          id: 'id-1',
          name: 'Olya',
          pile: NOT_EMPTY_PILE,
          tetro: {
            figure: LINE,
            row: 1,
            col: 4,
            color: RED,
          },
          score: 0,
          isGameOver: false,
        },
        'id-2': {
          id: 'id-2',
          name: 'Dasha',
          pile: EMPTY_PILE,
          tetro: {
            figure: GI,
            row: 0,
            col: 4,
            color: RED,
          },
          score: 0,
          isGameOver: false,
        },
      },
    },
    people: {},
    rooms: {},
    tetro: null,
  });

  expect(
    renderer
      .create(
        <Provider store={store}>
          <Others />
        </Provider>,
      )
      .toJSON(),
  ).toMatchSnapshot();
});
