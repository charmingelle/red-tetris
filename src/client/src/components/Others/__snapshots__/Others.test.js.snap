// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Others in room with 3 players 1`] = `
<ul
  className="others"
>
  <li
    className="other"
  >
    <div
      className="other-pile"
      style={
        Object {
          "height": "160px",
          "width": "80px",
        }
      }
    >
      <div
        className="square"
        style={
          Object {
            "backgroundColor": "#c00",
            "borderBottom": "1px solid #980000",
            "borderLeft": "1px solid #f00",
            "borderRight": "1px solid #980000",
            "borderTop": "1px solid  #f00",
            "height": "8px",
            "left": "0px",
            "top": "144px",
            "width": "8px",
          }
        }
      />
      <div
        className="square"
        style={
          Object {
            "backgroundColor": "#c00",
            "borderBottom": "1px solid #980000",
            "borderLeft": "1px solid #f00",
            "borderRight": "1px solid #980000",
            "borderTop": "1px solid  #f00",
            "height": "8px",
            "left": "8px",
            "top": "144px",
            "width": "8px",
          }
        }
      />
      <div
        className="square"
        style={
          Object {
            "backgroundColor": "#c00",
            "borderBottom": "1px solid #980000",
            "borderLeft": "1px solid #f00",
            "borderRight": "1px solid #980000",
            "borderTop": "1px solid  #f00",
            "height": "8px",
            "left": "16px",
            "top": "144px",
            "width": "8px",
          }
        }
      />
      <div
        className="square"
        style={
          Object {
            "backgroundColor": "#c00",
            "borderBottom": "1px solid #980000",
            "borderLeft": "1px solid #f00",
            "borderRight": "1px solid #980000",
            "borderTop": "1px solid  #f00",
            "height": "8px",
            "left": "16px",
            "top": "152px",
            "width": "8px",
          }
        }
      />
    </div>
    <div
      className="score"
    >
      Olya: 0
    </div>
  </li>
  <li
    className="other"
  >
    <div
      className="other-pile"
      style={
        Object {
          "height": "160px",
          "width": "80px",
        }
      }
    />
    <div
      className="score"
    >
      Dasha: 0
    </div>
  </li>
</ul>
`;
